### DISCUSSION TITLE
part-ii/2019-22/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#34] [Question comments]
Part II 2019 Automata and Formal Languages: Paper 1, Section I, H 

https://questions.tripos.org/part-ii/2019-22/

**Comment by callumhobbis** [comment]
<details>
<summary>Sketch solution</summary>

**(a):** Notes.

**(b):** (i) is a CFL, with CFG given by productions $S\to aSa\mid bSb\mid \varepsilon$.

(ii) is not a CFL. We use the pumping lemma on $u=0^p1^p=vwxyz$ where $p$ is the least prime with $p\geq n$, and by observing the length of $vw^kxy^kz$, we see we get an arithmetic progression of primes, which can't happen.

(iii) is a CFL with CFG given by productions $S\to a^4Sd^3|T$ and $T\to b^5Tc^2\mid\varepsilon$.

**(c):** Work with their CFGs directly, "merging" all the productions together whilst ensuring you don't get any collisions.

</details>

**Reply by thenewtonium** [reply]
Wdym by
> referring to their regular languages

(part **(c)**)?

**Reply by callumhobbis** [reply]
In lectures (lecture 14, 2021-22), the full proof of $LM$ being regular is just "$R_LR_M$ is regular and $\mathcal L(R_LR_M)=\mathcal L(R_L)\mathcal L(R_M)=LM$". It works perfectly well, but it hides all the hard work behind a previous result in lecture 13 where we show regular expressions define regular languages by drawing an actual $\varepsilon$-NFA, so I'm not sure if it's a sufficient answer for exam purposes.

**Reply by thenewtonium** [reply]
Um
Aren't we talking about CFLs and not regular languages in this question though??

**Reply by callumhobbis** [reply]
...very good point. That's my mistake, I'll edit that bit out so others don't get confused. 😂
